qt/aqt/browser.py

```diff
@@ -1210,27 +1217,34 @@ def setFilter(self, *args):
                 if i % 2 == 0:
                     txt += a + ":"
                 else:
-                    txt += re.sub(r"[*_\\]", r"\\\g<0>", a)
-                    for c in ' 　()"':
-                        if c in txt:
-                            txt = '"{}"'.format(txt.replace('"', '\\"'))
-                            break
+                    txt += re.sub(r'["*_\\]', r"\\\g<0>", a)
+                    txt = '"{}"'.format(txt.replace('"', '\\"'))
                     items.append(txt)
                     txt = ""
-            txt = " ".join(items)
-        # is there something to replace or append with?
-        if txt:
+            txt = " AND ".join(items)
+        try:
             if self.mw.app.keyboardModifiers() & Qt.AltModifier:
-                txt = "-" + txt
-            # is there something to replace or append to?
+                txt = self.col.backend.negate_search(txt)
```

Comment:
Ideally rsbackend.py will import (and thus re-export) required messages like ConcatenateSearchesIn, and pylib will provide simple wrappers so that users/GUI code doesn't need to call the backend directly - which gives us a bit more flexibility to change things around in the future. Looking through the codebase, I can see I haven't always managed to follow that rule myself however :-(